### PR TITLE
allow adding values with weights

### PR DESCRIPTION
**Pull request by githomin**
### What does this PR do?

allow adding values with weights

### Motivation

Final steps towards parity with sketches-java

### CODE REVIEW

**Comment by CharlesMasson** [comment]
I believe we need to remove the `int()` when computing the `rank` in `get_quantile_value`.

**Comment by githomin** [comment]
> I believe we need to remove the `int()` when computing the `rank` in `get_quantile_value`.

hrm, the tests say otherwise. can you think of a good test where that would come into play?

**Comment by CharlesMasson** [comment]
Maybe one for which the sum of weights is less than 1, which would make the rank be 1 no matter what the quantile is?

File: ddsketch/ddsketch.py [hunk 1]
```diff
@@ -122,19 +122,19 @@ def sum(self):
         """float: exact sum of the values added to the sketch"""
         return self._sum
 
-    def add(self, val):
+    def add(self, val, weight=1.0):
```
Comment (CharlesMasson) [inline ddsketch/ddsketch.py:125]: Should we validate that the `weight` is not negative?

**Review by CharlesMasson — APPROVED** [review]

**Pull request merged**